components/popover/popover-mixin.js

```diff
@@ -127,68 +215,375 @@ export const PopoverMixin = superclass => class extends superclass {
 		this._previousFocusableAncestor = getPreviousFocusableAncestor(this, false, false);
 
 		this._opener = getComposedActiveElement();
-		this._addAutoCloseHandlers();
+		this.#addAutoCloseHandlers();
+
+		await this.#position();
+
 		this._dismissibleId = setDismissible(() => this.close());
-		this._focusContent(this);
+
+		this.#focusContent(this);
+
+		this.#addRepositionHandlers();
+
 		this.dispatchEvent(new CustomEvent('d2l-popover-open', { bubbles: true, composed: true }));
+
+	}
+
+	renderPopover(content) {
+
+		const stylesMap = this.#getStyleMaps();
+		const widthStyle = stylesMap['width'];
+		const contentStyle = stylesMap['content'];
+
+		content = html`
+			<div class="content-width vdiff-target" style=${styleMap(widthStyle)}>
+				<div class="content-container" style=${styleMap(contentStyle)}>${content}</div>
+			</div>
+		`;
+
+		if (this._trapFocus) {
+			content = html`
+				<d2l-focus-trap @d2l-focus-trap-enter="${this.#handleFocusTrapEnter}" ?trap="${this._opened}">
+					${content}
+				</d2l-focus-trap>
+			`;
+		}
+
+		const positionStyles = {};
+		if (this._position) {
+			for (const prop in this._position) {
+				positionStyles[prop] = `${this._position[prop]}px`;
+			}
+		}
+
+		content = html`
+			<div class="content-position" style=${styleMap(positionStyles)}>
+				${content}
+			</div>
+		`;
+
+		const pointerPositionStyles = {};
+		if (this._pointerPosition) {
+			for (const prop in this._pointerPosition) {
+				pointerPositionStyles[prop] = `${this._pointerPosition[prop]}px`;
+			}
+		}
+
+		const pointer = !this._noPointer ? html`
+			<div class="pointer" style="${styleMap(pointerPositionStyles)}">
+				<div></div>
+			</div>
+		` : nothing;
+
+		return html`${content}${pointer}`;
+
+	}
+
+	async resize() {
+		if (!this._opened) return;
+		await this.#position();
 	}
 
 	toggleOpen(applyFocus = true) {
 		if (this._opened) return this.close();
 		else return this.open(!this._noAutoFocus && applyFocus);
 	}
 
-	_addAutoCloseHandlers() {
-		this.addEventListener('blur', this._handleAutoCloseFocus, { capture: true });
-		document.body.addEventListener('focus', this._handleAutoCloseFocus, { capture: true });
-		document.addEventListener('click', this._handleAutoCloseClick, { capture: true });
+	#handleAncestorMutationBound;
+	#handleAutoCloseClickBound;
+	#handleAutoCloseFocusBound;
+	#handleResizeBound;
+	#repositionBound;
+
+	#addAutoCloseHandlers() {
+		this.addEventListener('blur', this.#handleAutoCloseFocusBound, { capture: true });
+		document.body.addEventListener('focus', this.#handleAutoCloseFocusBound, { capture: true });
+		document.addEventListener('click', this.#handleAutoCloseClickBound, { capture: true });
 	}
 
-	_clearDismissible() {
+	#addRepositionHandlers() {
```

Comment:
This code is basically copied from `DropdownContentMixin`. The only real difference is that we need to handle the opener `IntersectionObserver` here, since the opener could literally be any element, as compared to dropdown, where we do this wire-up in `DropdownOpenerMixin`.